src/main/daml/Daml/Finance/Instrument/Generic/Util.daml

```diff
@@ -175,21 +178,81 @@ calculateClaimsFromFixSwapStream s periodicSchedule streamSchedule useAdjustedDa
   let f = fromSome s.calculationPeriodAmount.calculation.fixedRateSchedule
   pure $ createFixRatePaymentClaims streamSchedule periodicSchedule useAdjustedDatesForDcf f.initialValue issuerPaysLeg s.calculationPeriodAmount.calculation.dayCountFraction currency
 
+-- | Rule when one specific floating rate is provided for a stub period.
+getSingleStubRateFromFpml floatingRate = do
+  Some (Observe floatingRate.floatingRateIndex)
+
+-- | Rule when two specific floating rates are provided for a stub period.
```

Comment:
For the sake of clarity, I would simplify the function description to

```
-- | Linearly Interpolates two rates within a period. as specified in https://www.isda.org/a/aWkgE/Linear-interpolation-04022022.pdf
```
The reason for my comment is the fact that `Rule` has a rather distinct meaning within Daml jargon (specifically referring to the `facts` vs `rules` design pattern).